### PR TITLE
dev/sg: add 1pass to sg setup

### PR DESCRIPTION
**Pull request by bobheadxi**
Installation follows the steps in https://developer.1password.com/docs/cli/get-started/#install , then logs in to https://team-sourcegraph.1password.com/

## Test plan

Only tested the MacOS path, the linux installation looks scary.

image

### CODE REVIEW

**Comment by bobheadxi** [comment]
Current dependencies on/for this PR:
* main
  * **PR #36335** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/36335" target="_blank">Graphite</a>  👈
    * **PR #36336** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/36336" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/36335?utm_source=stack-comment).

**Review by mrnugget — COMMENTED** [review]
(OMG! I just noticed I never submitted my review... Well, here's some post-merge comments then 😄 )

File: dev/sg/sg_setup_mac.go [hunk 1]
```diff
@@ -289,6 +289,15 @@ YOU NEED TO RESTART 'sg setup' AFTER RUNNING THIS COMMAND!`,
 		autoFixing: true,
 		dependencies: []*dependency{
 			dependencyGcloud,
+			{
+				name:          "1password",
+				onlyTeammates: true,
+				instructionsCommands: `
+brew install --cask 1password/tap/1password-cli
```
[the remark this answers is not in the document]
Comment (bobheadxi) [reply dev/sg/sg_setup_mac.go:296]: Ah shoot it doesn't, I thought I tried this but it must've skipped over - working on a fix

[the remark this answers is not in the document]
Comment (bobheadxi) [reply dev/sg/sg_setup_mac.go:296]: Is it intentional for fixes to not accept prompts?

**Comment by bobheadxi** [comment]
Review comments applied in https://github.com/sourcegraph/sourcegraph/pull/36489 !

File: dev/sg/sg_setup_mac.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrnugget) [reply dev/sg/sg_setup_mac.go:296]: I don't think it's intentional in the "we don't want to show prompts"-way, but it's due to `output` rendering things and taking over output.

[the remark this answers is not in the document]
Comment (mrnugget) [reply dev/sg/sg_setup_mac.go:296]: IIRC a Caddy command also requires you to type prompt in, but that worked in `sg setup`.

[the remark this answers is not in the document]
Comment (bobheadxi) [reply dev/sg/sg_setup_mac.go:296]: In https://github.com/sourcegraph/sourcegraph/pull/36489 I just pipe `os.Stdin` and it seems to work 🤔

[the remark this answers is not in the document]
Comment (mrnugget) [reply dev/sg/sg_setup_mac.go:296]: Fine with me!

**Pull request merged**